core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java

```diff
@@ -280,6 +280,10 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                     final boolean endOfStream = contentEmpty && !transferEncodingChunked;
                     this.req = req = DecodedHttpRequest.of(endOfStream, eventLoop, id, 1, headers,
                                                            keepAlive, inboundTrafficController, routingCtx);
+                    final long maxRequestLength = req.maxRequestLength();
+                    if (maxRequestLength > 0 && contentLength > maxRequestLength) {
+                        abortLargeRequest(ctx, req, id, endOfStream, keepAliveHandler, true);
+                    }
```

Comment:
I am unsure if it is necessary to pass the content-length exceeding request to the service chains. Should we `fail()` the request instead? What do you think @minwoox?

Reply:
I wasn't worried about performance, but I was considering the race condition between the response sent by the user and the aborted response.

That case would be rare, so I think it can be ignored.

Reply:
Yeah, let's revisit this if we encounter any issue with this. 😉